File: ingeneo_technical_test/ingeneo_technical_test/src/main/java/com/example/ingeneo_technical_test/entity/dto/WarehouseDTO.java
package com.example.ingeneo_technical_test.entity.dto;

import com.example.ingeneo_technical_test.entity.Port;
import com.example.ingeneo_technical_test.entity.Store;
import com.example.ingeneo_technical_test.enumerations.WarehouseType;
import com.fasterxml.jackson.annotation.JsonInclude;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author Johan Casagua
 */
@Data @AllArgsConstructor @NoArgsConstructor @Builder
@JsonInclude(JsonInclude.Include.NON_NULL)
public class WarehouseDTO {
	private Long id;
	private String name;
	private String location;
	private String storageCapacity;
    private WarehouseType type; 
	private Port port;
	private Store store;
}

File: frontend/ingeneott/src/app/models/warehouse.dto.ts
import { Port } from "./port.dto";
import { Store } from "./store.dto";

export class WarehouseDTO {
    id: number;
    name: string;
    location: string;
    storageCapacity: string;
    type: WarehouseType;
    port: Port;
    store: Store;
}

export enum WarehouseType {
    STORE = 'STORE',
    PORT = 'PORT'
}